### PR TITLE
Fix bug in DPU Platform Info

### PR DESCRIPTION
**Pull request by SalDaniele**
We need to create a new instance of the PlatformInfo struct to ensure the NewIntelDetector is added.

Additionally, fix a small bug in determining a device is an Intel DPU. The Intel IPU pci Class/Vendor/Product should be:

Network controller Intel Corporation Infrastructure Data Path Function

### CODE REVIEW

**Comment by SalDaniele** [comment]
@bn222 PTAL

File: internal/platform/platform.go [hunk 1]
```diff
@@ -74,7 +88,25 @@ func (pi *PlatformInfo) VspPlugin(dpuMode bool) (*plugin.GrpcPlugin, error) {
...
 			return nil, fmt.Errorf("Failed to detect any DPU devices")
 		}
-		// FIXME: We need to find a good way to determine VF from PF with the Intel DPU versus different DPU vendors.
-		return detectors[0].VspPlugin(dpuMode), nil
+
+		var pfDevices []*ghw.PCIDevice
+		var pfDetectors []VendorDetector
+
+		for i, device := range dpuDevices {
+			isVF, err := isVirtualFunction(device.Address)
```
Comment (wizhaoredhat) [inline internal/platform/platform.go:96]: We might need this logic in listDpuDevices instead...

File: internal/platform/platform.go [hunk 2]
```diff
@@ -63,6 +65,18 @@ func (pi *PlatformInfo) listDpuDevices() ([]ghw.PCIDevice, []VendorDetector, err
 	return dpuDevices, activeDetectors, nil
 }
 
+func isVirtualFunction(device string) (bool, error) {
```
Comment (wizhaoredhat) [inline internal/platform/platform.go:68]: I think this belongs to ipu.go.

**Comment by wizhaoredhat** [comment]
since Sal is on PTO. Let's merge this in and fix the issues above in a seperate PR.

**Pull request merged**